### ISSUE TITLE
Conflict with Pace.js

### ISSUE THREAD

**Issue by c0mrx** [barbajs/barba#391]
## The problem

Barba will not complete a transition with Pace.js active.

## Description

On the newest build available (2.3.9), the `leave` function will run as expected, but will hang as soon as it tried to run the `enter` function. The page is not reloaded and the console shows no errors. Pace also seems to run fine and sees the activity. Oddly enough, on the 2.0.1 build of Barba, everything works as expected and Pace does not break the transition function.

### Expected behaviour

The `leave` and `enter` events should fire as expected.

### Actual behaviour

The `leave` function runs but the `enter` function does not. There are no errors in the console, and the page is not refreshed. Disabling Pace resolves the issue.

### Code To Reproduce Issue [ Good To Have ]

Please note that Pace.js should be active in the project to reproduce this bug. The transition code I've used is as follows:

```

barba.init( {
  debug: true,
  prefetchIgnore: true,
  transitions: [ {
    sync: false,
    leave( {
      current,
      next,
      trigger
    } ) {

      console.log( "leave", current, next, trigger );
      // Do something with `current.container` for your leave transition
      // then return a promise or use `this.async()`
      return new Promise( resolve => {

        var target = current.container;

        anime( {
          targets: target,
          // translateX: [ '0%', '-35%' ],
          opacity: [ 1, 0 ],
          duration: 300,
          easing: 'linear',
          complete: () => {
            resolve();
            // mainScrollbar.setPosition( 0, 0 );
          },
        } );

      } );
    },
    enter( {
      current,
      next,
      trigger
    } ) {
      // Do something with `next.container` for your enter transition
      // then return a promise or use `this.async()`

      console.log( "enter", current, next, trigger );

      return new Promise( resolve => {

        var target = next.container;

        anime( {
          targets: target,
          // translateX: [ '35%', '0%' ],
          opacity: [ 0, 1 ],
          duration: 300,
          delay: 100,
          easing: 'linear',
          complete: () => {
            resolve();
          },
        } );

      } );
    },
  }, ],
} );
```

### Steps to reproduce issue [ Good To Have ]

I'm using a Wordpress instance with Pace.js and Barba (2.3.9). I've tried both Local and Live environments and the problem persists with both.

---

## Relevant Links

[Pace.js GitHub repo](https://github.com/HubSpot/pace)

## Environment

- Barba version that exhibits the issue: any build above 2.0.1
- Build used:
  - [x] browser (dist/)
  - [ ] module (esm/)
  - [ ] main (lib/)
- Browser OS/version used:
Google Chrome (Version 74.0.3729.131)
MacOS 10.4.4

**Comment by bezgodov** [comment]
Facing same issue

**Comment by luruke** [comment]
Thanks @c0mrx for the issue!
@thierrymichel do you have any idea what could have changed between `2.0.1` and `2.3.9` that broke pace.js?

Considering that library isn't updated since 2015, I would exclude big change to support it...

**Comment by thierrymichel** [comment]
@luruke probably switching from `fetch` to `XMLHttpRequest`… :(

**Comment by thierrymichel** [comment]
It's harder to debug than expected BUT … 
1. it works fine with the [non-minified version](https://raw.githubusercontent.com/HubSpot/pace/v1.0.2/pace.js) 🤔 
2. with my own minified version, transitions works well, but `pace` first load not (slow progress bar, not removed). 🙃

ping @c0mrx

**Comment by c0mrx** [comment]
@thierrymichel Interesting,

I just realized I was using the non-minified version, but since we use Gulp to transpile/minify, I'm thinking pace might have been broken in that process. The project I was using Pace on has launched and I just ended up getting rid of pace, as the site was pretty quick naturally (thanks, barba! 😄).

I plan to use barba in my next project I should be starting in a few days. I'll test out pace again and report back.

@bezgodov Are you using the non-minified version? This would be helpful to know.

Thanks!

**Comment by thierrymichel** [comment]
FYI minified version using mix w/ webpack (don’t ask me why 😄) gives another behaviour...

**Comment by bezgodov** [comment]
> @bezgodov Are you using the non-minified version? This would be helpful to know.

No, I was using minified version (I even tried to include them inline but with no results)

**Comment by c0mrx** [comment]
Okay, so I've tried both the minified and non-minified versions. Both do nothing once a link is clicked (no transition happens), but Pace still runs for some strange reason. It's like it knows theres a request happening but it stops everything else from happening. I'm really stumped on this one.

I've just started refraining from using Pace as it's an old library anyways and there are other ways of telling a user something is happening (with subtle animation or other indicators). I don't think it's worth trying to accommodate this one plugin, because everything else works great from my experience.

I just wanted to say thanks for looking into it a little more @thierrymichel ! I'll leave this open for further discussion if necessary. 😄
